File: backend-crud/SerializationService/JsonSerializationService.fs
namespace RebelSoftware.SerializationService

open System

open Newtonsoft.Json

module Serialization =
    
    type SerializationService<'a> = { //todo: rename to TypeSerializer?
        DeserializeFromJson : string -> 'a; //todo: generic type param
        SerializeToJson : obj -> string;
    }

    let private deserializeFromJson<'a> (jsonString:string) = 
        let employee = JsonConvert.DeserializeObject<'a>(jsonString)
        employee

    let private serializeToJson object =
        let jsonString = JsonConvert.SerializeObject object
        jsonString

    let createSerializationService<'a> () = { 
        DeserializeFromJson = deserializeFromJson<'a>;
        SerializeToJson = serializeToJson;
    }

File: backend-crud-CSharp/SerializationService/JsonSerializationService.cs
using System;
using Newtonsoft.Json;

namespace RebelSoftware.Serialization
{
    public class JsonSerializationService : ISerializationService
    {
        public T DeserializeFromJson<T>(string jsonString)
        {
            var retVal = JsonConvert.DeserializeObject<T>(jsonString);
            return retVal;
        }

        public string SerializeToJson<T>(T objToSerialize)
        {
            var retVal = JsonConvert.SerializeObject(objToSerialize);
            return retVal;
        }
    }
}
